src/relay/pass/quantize_rewrite.cc

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file quantize_rewrite.cc
+ * \brief Lower quantized ops to exisiting Relay ops.
+ */
+
+#include <tvm/relay/analysis.h>
+#include <tvm/relay/attrs/qnn.h>
+#include <tvm/relay/op_attr_types.h>
+#include <tvm/relay/quantize_util.h>
+#include <tvm/relay/transform.h>
+#include "pattern_util.h"
+
+namespace tvm {
+namespace relay {
+
+Expr QuantizeForwardRewrite(const Call& ref_call, const Array<Expr>& new_args, const NodeRef& ctx) {
+  CHECK_EQ(new_args.size(), 1);
+  Expr data = new_args[0];
+  const auto* attrs = ref_call->attrs.as<QuantizeAttrs>();
+  const auto out_dtype = attrs->out_dtype;
+  const auto* new_tensor = data.operator->()->checked_type().as<TensorTypeNode>();
+  CHECK(new_tensor) << "Expected TensorTypeNode but was " << data.operator->()->checked_type();
+  const auto output_zero_point = MakeConstantScalar(Int(32), attrs->output_zero_point);
+  const auto scale = MakeConstantScalar(Float(32), attrs->output_scale);
+  const int32_t min_val = get_qmin(out_dtype);
+  const int32_t max_val = get_qmax(out_dtype);
+  auto scale_data = Cast(Round(Divide(data, scale)), Int(32));
```

Comment:
As discussed in the RFC, this should be round_away_from_zero, i.e. std::round in C++. How about this Round here? I don't see definition of this Round.

Reply:
This is Relay Round operator which translates to LLVM round intrinsic

https://llvm.org/docs/LangRef.html#llvm-round-intrinsic

The comment on the above link says that it has same behavior as LIBM round function which is defined here
https://sourceware.org/newlib/libm.html#lround

This is round-away-from-zero.

Reply:
Ok. That's make sense.